app/instance-initializers/posthog.js

```diff
@@ -0,0 +1,14 @@
+import config from 'codecrafters-frontend/config/environment';
+import posthog from 'posthog-js';
+
+export function initialize() {
+  if (typeof FastBoot === 'undefined') {
+    if (config.environment === 'production') {
```

Comment:
@libmartinito a bit of a weird pattern to have two nested ifs like this, it's more common to either do early returns, or combine the conditions using `&&`